src/axom/quest/ArrayIndexer.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
```

Comment:
Is this class generally useful? In particular, should it go into core, or should the functionality be made part of axom::ArrayView?

Reply:
I think we should definitely discuss that, particularly if we want to support blueprint structured meshes.  We may not want to expose it to users until we'ready to commit to using it in the Array classes.  I'll open an issue.

Reply:
Issue https://github.com/LLNL/axom/issues/1216

Reply:
Will do.